### PR TITLE
Fix time (and date) fields editing in forms

### LINKED ISSUE #33000: Date field values are shown as NULL in attribute table form view
I have a layer with a time field
image
The values display in attribute table, in table view
image
not in form view
image

Version de QGIS | 3.10.0-A Coruña | Révision du code | 07378a71cb
-- | -- | -- | --
Compilé avec Qt | 5.11.2 | Utilisant Qt | 5.11.2
Compilé avec GDAL/OGR | 3.1.0dev | Utilisé avec GDAL/OGR | 3.1.0dev
Compilé avec GEOS | 3.8.0-CAPI-1.13.1 | Utilisé avec GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
Version du client PostgreSQL | 11.5 | Version de SpatiaLite | 4.3.0
Version de QWT | 6.1.3 | Version de QScintilla2 | 2.10.8
Compilé avec PROJ | 7.0.0 | Fonctionne avec PROJ | Rel. 7.0.0, March 1st, 2020
OS Version | Windows 10 (10.0) | Cette copie de QGIS dispose d'une sortie de débogage.
Extensions Python actives | arrayplus; DissolveWithStats; firstaid; GroupStats; MapsPrinter; MapsPrinter-add_batch_export; pluginbuilder3; PluginLoadTimes; plugin_reloader; Qgis2threejs; db_manager; MetaSearch; processing

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #33000

### CODE REVIEW

File: src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp [hunk 1]
```diff
@@ -212,12 +212,19 @@ void QgsDateTimeEditWrapper::updateValues( const QVariant &value, const QVariant
...
   QDateTime dateTime;
+
   switch ( field().type() )
   {
     case QVariant::DateTime:
+      dateTime = value.toDateTime();
+      break;
     case QVariant::Date:
+      dateTime.setDate( value.toDate() );
+      dateTime.setTime( QTime::currentTime() );
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp]: Not 00:00?

File: tests/src/gui/testqgsdatetimeedit.cpp [hunk 2]
```diff
@@ -226,5 +235,31 @@ void TestQgsDateTimeEdit::focus()
   QCOMPARE( dateedit3->text(), QStringLiteral( "nope" ) );
 }
 
+void TestQgsDateTimeEdit::testDateTime()
+{
+  QgsApplication::setNullRepresentation( QString( "nope" ) );
+  QWidget *w = new QWidget(); //required for focus events
```
Comment (m-kuhn) [inline tests/src/gui/testqgsdatetimeedit.cpp]: Isn't this leaked?
I think it's good practice to not leak in tests, so they can be run through valgrind to find issues.

Comment (elpaso) [reply tests/src/gui/testqgsdatetimeedit.cpp]: Good catch, I just copied that init code from the existing test case a few lines above and focused on the test logic, funny that it escaped from two reviewers: https://github.com/qgis/QGIS/pull/32158, you really do have an eye for this!

File: src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp (hunk 1 above)
Comment (elpaso) [reply src/gui/editorwidgets/qgsdatetimeeditwrapper.cpp]: It doesn't really matter, you can set it to whatever you want, it won't be seen or saved (or at least that's the idea).

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**